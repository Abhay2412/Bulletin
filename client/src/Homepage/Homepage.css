.Homepage {
  text-align: left;
}

.Bulletin-logo {
  position: absolute;
  width: 53px;
  height: 54px;
  left: calc((100vw / 4) - 100px);
  top: 192px;
}

.Homepage-background {
  background-color: white;
  min-height: 150vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.primary {
  position: absolute;
  width: 172px;
  height: 59px;
  left: calc((100vw / 4) - 47px);
  top: 160px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 50px;
  line-height: 59px;

  color: #534029;
}

.secondary {
  position: absolute;
  width: 690px;
  height: 23px;
  left: calc((100vw / 4) - 100px);
  top: 230px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;

  color: #6d5a43;
}

.button {
  position: absolute;
  width: 178px;
  height: 38px;
  left: calc((100vw / 4) + 257px);
  top: 287px;

  border: 1px solid #f0e6db;
  background-color: #ecd1c5;
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

  color: white;
  text-align: center;
  font-size: 20px;
  font-weight: normal;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;

  opacity: 0.8;
  transition: 0.3s;
  cursor: pointer;
}

.button:hover {
  opacity: 1;
}

.button:active {
  background-color: #ecd1c2;
  box-shadow: 0 1px #666;
  transform: translateY(3px);
}

.input[type="text"] {
  position: absolute;
  width: 342px;
  height: 38px;
  left: calc((100vw / 4) - 100px);
  top: 287px;

  box-sizing: border-box;
  border: 0px;

  background: #ffffff;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

  color: #9082708F;
  font-size: 20px;
  font-weight: 600;
  font-style: normal;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;

  -webkit-transition: 0.5s;
  transition: 0.5s;
  outline: none;
}

.input[type="text"]:focus {
  border: 2px solid #ecd1c2;
}
