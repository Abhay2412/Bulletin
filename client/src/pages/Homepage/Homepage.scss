.email-signup-page {
  width: 100%;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;

  .email-signup-page-inner {
    margin-top: 7%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .Bulletin-logo {
    width: 53px;
    height: 54px;
  }

  .title-container {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
  }

  .email-signup-primary {
    font-style: normal;
    font-weight: bold;
    font-size: 50px;
    color: #534029;
  }

  .secondary {
    width: 690px;
    font-style: normal;
    font-weight: bold;
    font-size: 20px;
    line-height: 23px;

    color: #6d5a43;
  }

  .email-signup-button {
    border: 1px solid #f0e6db;
    background-color: #ecd1c5;
    box-sizing: border-box;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

    color: white;
    text-align: center;
    font-size: 20px;
    font-weight: normal;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
      Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;

    opacity: 0.8;
    transition: 0.3s;
    cursor: pointer;
  }

  .email-signup-button:hover {
    opacity: 1;
  }

  .email-signup-button:active {
    background-color: #ecd1c2;
    box-shadow: 0 1px #666;
    transform: translateY(3px);
  }

  .email-input-field {
    box-sizing: border-box;
    border: 0px;
    width: 40%;
    margin-right: 30px;

    background: #ffffff;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

    color: #9082708f;
    font-size: 20px;
    font-weight: 600;
    font-style: normal;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
      Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;

    -webkit-transition: 0.5s;
    transition: 0.5s;
    outline: none;
  }

  .email-input-field:focus {
    border: 2px solid #ecd1c2;
  }
}
