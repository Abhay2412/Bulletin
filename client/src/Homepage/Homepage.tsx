import React from "react";
import logo from "./logo.svg";
import "./Homepage.css";

const Homepage = () => {

    return (
        <div className="Homepage">
            <header className="Homepage-background">
                <img src={logo} className="Bulletin-logo"></img>
                <h1 className="primary">Bulletin</h1>
                <h3 className="secondary">Organize your household and keep everyone in the loop with Bulletin</h3>
                <button className="button">Sign Up Now!</button>
                <form>
                    <input type="text" id="email" placeholder="Email" className="input"></input>
                </form>
            </header>
        </div>
    );
};


export default Homepage;