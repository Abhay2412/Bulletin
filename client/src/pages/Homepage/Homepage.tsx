import React, { useEffect, useState } from "react";
import logo from "./logo.svg";
import { Button, TextField } from "@material-ui/core";
import "./Homepage.scss";
import { useNavigate } from "react-router-dom";
import { useTypedSelector } from "../../hooks/ReduxHooks";
import { selectUserData } from "../../actions/UserActions/UserSelector";

const Homepage = () => {
  const navigate = useNavigate();
  const userData = useTypedSelector(selectUserData);
  const [email, setEmail] = useState("");
  const [errors, setErrors] = useState("");

  /**
   * Handles the email field.
   * @param event 
   */
  const handleEmailChange = (
    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setEmail(event.target.value);
  };

  /**
  * Allows the validation of the value in the email field.
  */
  const validateData = () => {
    if (!email) {
      setErrors("Please enter an email.");
      return false;
    } else if (
      !/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(
        email
      )
    ) {
      setErrors("Please enter a valid email.");
      return false;
    }

    return true;
  };

  /**
   * Handles what happens when the "sign up now" butoon is pressed.
   */
  const handleSignupClick = () => {
    if (!validateData()) {
      return;
    }
    navigate("/signup/" + email);
  };

  useEffect(() => {
    if (userData?.id && localStorage.getItem("refresh")) {
      navigate("/boardsView");
    }
  }, []);

  return (
    <div className="email-signup-page">
      <div className="email-signup-page-inner">
        <div className="title-container">
          <div>
            <img src={logo} className="Bulletin-logo"></img>
          </div>
          <div>
            <h1 className="email-signup-primary">Bulletin</h1>
          </div>
        </div>
        <div className="title-container">
          <div>
            <h3 className="secondary">
              Organize your household and keep everyone in the loop with
              Bulletin
            </h3>
          </div>
        </div>
        <div className="title-container">
          <TextField
            id="homepage-email-field"
            className="email-input-field"
            InputLabelProps={{ shrink: false }}
            value={email}
            onChange={handleEmailChange}
            label={email === "" ? "Email" : ""}
            variant="outlined"
            type="email"
            error={errors !== ""}
            helperText={errors}
          />
          <Button className="email-signup-button" onClick={handleSignupClick}>
            Sign Up Now!
          </Button>
        </div>
      </div>
    </div>
  );
};

export default Homepage;
