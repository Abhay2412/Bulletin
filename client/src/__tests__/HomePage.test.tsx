import React from "react";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import Homepage from "../pages/Homepage/Homepage";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("testing <HomePage /> component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Homepage />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should change email field value", () => {
    wrapper
      .find("input")
      .find("#homepage-email-field")
      .simulate("change", {
        target: { value: "test@domain.com" },
      });
    expect(
      wrapper.find("input").find("#homepage-email-field").get(0).props.value
    ).toEqual("test@domain.com");
  });

  it("should have email error", () => {
    wrapper
      .find("input")
      .find("#homepage-email-field")
      .simulate("change", {
        target: { value: "" },
      });

    wrapper.find("button").find(".email-signup-button").simulate("click");
    expect(
      wrapper.find("p").find("#homepage-email-field-helper-text").text()
    ).toEqual("Please enter an email.");
  });
});
