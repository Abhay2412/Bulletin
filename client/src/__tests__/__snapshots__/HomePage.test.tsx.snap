// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <HomePage /> component should render properly 1`] = `
<div
  className="email-signup-page"
>
  <div
    className="email-signup-page-inner"
  >
    <div
      className="title-container"
    >
      <div>
        <img
          className="Bulletin-logo"
          src="logo.svg"
        />
      </div>
      <div>
        <h1
          className="email-signup-primary"
        >
          Bulletin
        </h1>
      </div>
    </div>
    <div
      className="title-container"
    >
      <div>
        <h3
          className="secondary"
        >
          Organize your household and keep everyone in the loop with Bulletin
        </h3>
      </div>
    </div>
    <div
      className="title-container"
    >
      <div
        className="MuiFormControl-root MuiTextField-root email-input-field"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="homepage-email-field"
          id="homepage-email-field-label"
        >
          Email
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="homepage-email-field"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="email"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text email-signup-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sign Up Now!
        </span>
      </button>
    </div>
  </div>
</div>
`;
